src/main/scala/com/gu/scanamo/Table.scala

```diff
@@ -61,6 +61,103 @@ case class Table[V: DynamoFormat](name: String) {
   def get(key: UniqueKey[_]) = ScanamoFree.get[V](name)(key)
   def getAll(keys: UniqueKeys[_]) = ScanamoFree.getAll[V](name)(keys)
   def delete(key: UniqueKey[_]) = ScanamoFree.delete(name)(key)
+
+  /**
+    * Performs the chained operation if the condition is met
+    *
+    * {{{
+    * >>> case class Farm(animals: List[String])
+    * >>> case class Farmer(name: String, age: Long, farm: Farm)
+    *
+    * >>> import com.gu.scanamo.syntax._
+    * >>> import com.gu.scanamo.query._
+    * >>> import com.amazonaws.services.dynamodbv2.model.ScalarAttributeType._
+    * >>> val client = LocalDynamoDB.client()
+    *
+    * >>> val farmersTable = Table[Farmer]("nursery-farmers")
+    * >>> LocalDynamoDB.withTable(client)("nursery-farmers")('name -> S) {
+    * ...   val farmerOps = for {
+    * ...     _ <- farmersTable.put(Farmer("McDonald", 156L, Farm(List("sheep", "cow"))))
+    * ...     _ <- farmersTable.given('age -> 156L).put(Farmer("McDonald", 156L, Farm(List("sheep", "chicken"))))
+    * ...     _ <- farmersTable.given('age -> 15L).put(Farmer("McDonald", 156L, Farm(List("gnu", "chicken"))))
+    * ...     farmerWithNewStock <- farmersTable.get('name -> "McDonald")
+    * ...   } yield farmerWithNewStock
+    * ...   Scanamo.exec(client)(farmerOps)
+    * ... }
+    * Some(Right(Farmer(McDonald,156,Farm(List(sheep, chicken)))))
+    *
+    * >>> case class Thing(a: String, maybe: Option[Int])
+    * >>> val thingTable = Table[Thing]("things")
+    * >>> LocalDynamoDB.withTable(client)("things")('a -> S) {
+    * ...   val ops = for {
+    * ...     _ <- thingTable.putAll(List(Thing("a", None), Thing("b", Some(1)), Thing("c", None)))
+    * ...     _ <- thingTable.given(attributeExists('maybe)).put(Thing("a", Some(2)))
+    * ...     _ <- thingTable.given(attributeExists('maybe)).put(Thing("b", Some(3)))
+    * ...     _ <- thingTable.given(Not(attributeExists('maybe))).put(Thing("c", Some(42)))
+    * ...     _ <- thingTable.given(Not(attributeExists('maybe))).put(Thing("b", Some(42)))
+    * ...     things <- thingTable.scan()
+    * ...   } yield things
+    * ...   Scanamo.exec(client)(ops).toList
+    * ... }
+    * List(Right(Thing(b,Some(3))), Right(Thing(c,Some(42))), Right(Thing(a,None)))
+    *
+    * >>> case class Compound(a: String, maybe: Option[Int])
+    * >>> val compoundTable = Table[Compound]("compounds")
+    * >>> LocalDynamoDB.withTable(client)("compounds")('a -> S) {
+    * ...   val ops = for {
+    * ...     _ <- compoundTable.putAll(List(Compound("alpha", None), Compound("beta", Some(1)), Compound("gamma", None)))
+    * ...     _ <- compoundTable.given(attributeExists('maybe) and 'a -> "alpha").put(Compound("alpha", Some(2)))
+    * ...     _ <- compoundTable.given(attributeExists('maybe) and 'a -> "beta").put(Compound("beta", Some(3)))
+    * ...     _ <- compoundTable.given(Condition('a -> "gamma") and attributeExists('maybe)).put(Compound("gamma", Some(42)))
```

Comment:
Having to use `Condition()` if it's the first part of a compound condition is ugly. I'm hoping to find a better alternative, but I this is just about good enough to put out there.